src/Microsoft.AspNet.Http.Features/IHttpResponseFeature.cs

```diff
@@ -13,6 +13,7 @@ public interface IHttpResponseFeature
     {
         int StatusCode { get; set; }
         string ReasonPhrase { get; set; }
+        long? ContentLength { get; set; }
```

Comment:
Takes the allocations to zero if the ContentLength is maintained as the source of the header and the header set/get is intercepted; rather than otherway around.